backend/src/api/database-migration.ts

```diff
@@ -0,0 +1,176 @@
+import config from '../config';
+import { DB } from '../database';
+import logger from '../logger';
+
+class DatabaseMigration {
+  private static currentVersion = 1;
+  private queryTimeout = 120000;
+
+  constructor() { }
+
+  public async $initializeOrMigrateDatabase(): Promise<void> {
+    if (!await this.$checkIfTableExists('statistics')) {
+      const connection = await DB.pool.getConnection();
+      for (const query of this.getInitializeTableQueries()) {
+        await connection.query<any>({ sql: query, timeout: this.queryTimeout });
+      }
+      connection.release();
+      logger.info(`Initial database tables has been created`);
+    }
+
+    if (await this.$checkIfTableExists('state')) {
+      const databaseSchemaVersion = await this.$getSchemaVersionFromDatabase();
+      if (DatabaseMigration.currentVersion > databaseSchemaVersion) {
+        const connection = await DB.pool.getConnection();
+        for (const query of this.getMigrationQueriesFromVersion(databaseSchemaVersion)) {
+          await connection.query<any>({ sql: query, timeout: this.queryTimeout });
+        }
+        connection.release();
+        this.$updateToLatestSchemaVersion();
+        logger.info(`Database schema has been migrated to the latest version (${DatabaseMigration.currentVersion})`);
+      }
+    } else {
+      const connection = await DB.pool.getConnection();
+      for (const query of this.getMigrationQueriesFromVersion(0)) {
+        await connection.query<any>({ sql: query, timeout: this.queryTimeout });
+      }
+      connection.release();
+      this.$updateToLatestSchemaVersion();
+      logger.info(`Database schema has been migrated to the latest version (${DatabaseMigration.currentVersion})`);
+    }
+  }
+
+  private async $getSchemaVersionFromDatabase(): Promise<number> {
+    const connection = await DB.pool.getConnection();
+    const query = `SELECT number FROM state WHERE name = 'schema_version';`;
```

Comment:
@softsimon SQL queries should be imported as constants so it's easier to maintain and track  vs sql changes